testsuite/pytests/test_compartmental_model.py

```diff
@@ -0,0 +1,823 @@
+                                    "meaning that no compartments have been added."):
+            nest.Simulate(10.)
+
+        # test connection port out of range for current input
+        n_neat = nest.Create('cm_default')
+        nest.SetStatus(n_neat, {"compartments": {"parent_idx": -1, "params": SP}})
+        nest.SetStatus(n_neat, {"compartments": {"parent_idx": 0, "params": SP}})
+        dc = nest.Create('dc_generator', {'amplitude': 2.0})
+
+        with self.assertRaisesRegex(nest.kernel.NESTError,
+                                    "UnknownPort in SLI function Connect_g_g_D_D: "
+                                    "Port with id 3 does not exist. Valid current "
+                                    "receptor ports for cm_default are in \[0, 2\[."):
+            nest.Connect(dc, n_neat, syn_spec={'synapse_model': 'static_synapse', 'weight': 1.,
+                                                 'receptor_type': 3})
+
+        # test connection port out of range for spike input
+        n_neat = nest.Create('cm_default')
+        nest.SetStatus(n_neat, {"compartments": {"parent_idx": -1, "params": SP}})
+        nest.SetStatus(n_neat, {"receptors": {"comp_idx": 0, "receptor_type": "GABA"}})
+        nest.SetStatus(n_neat, {"receptors": {"comp_idx": 0, "receptor_type": "GABA"}})
+        nest.SetStatus(n_neat, {"receptors": {"comp_idx": 0, "receptor_type": "GABA"}})
+        sg = nest.Create('spike_generator', 1, {'spike_times': [10.]})
+
+        with self.assertRaisesRegex(nest.kernel.NESTError,
+                                    "UnknownPort in SLI function Connect_g_g_D_D: "
+                                    "Port with id 3 does not exist. Valid spike "
+                                    "receptor ports for cm_default are in \[0, 3\[."):
+            nest.Connect(sg, n_neat, syn_spec={'synapse_model': 'static_synapse', 'weight': 1.,
+                                                 'receptor_type': 3})
+
+
+        # test connection with unknown recordable
+        n_neat = nest.Create('cm_default')
+        nest.SetStatus(n_neat, {"compartments": {"parent_idx": -1, "params": SP}})
+        mm = nest.Create('multimeter', 1, {'record_from': ['v_comp1'], 'interval': 1.})
+
+        with self.assertRaisesRegex(nest.kernel.NESTError,
+                                    "IllegalConnection in SLI function Connect_g_g_D_D: "
+                                    "Creation of connection is not possible because:\n"
+                                    "Cannot connect with unknown recordable v_comp1"):
+            nest.Connect(mm, n_neat)
+
+
+def suite():
+    # makeSuite is sort of obsolete http://bugs.python.org/issue2721
+    # using loadTestsFromTestCase instead.
+    suite = unittest.TestLoader().loadTestsFromTestCase(NEASTTestCase)
+    return unittest.TestSuite([suite])
+
+
+def run():
+    runner = unittest.TextTestRunner(verbosity=2)
+    runner.run(suite())
+
+
+if __name__ == "__main__":
+    run()
```

Comment:
Please add a newline to the end of the file

Reply:
Added the lines